osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -331,11 +334,15 @@ private void updateBonusScore()
 
             while (completedFullSpins.Value != spins)
             {
-                var tick = ticks.FirstOrDefault(t => !t.Result.HasResult);
+                var tick = ticks.FirstOrDefault(t => !t.Result.HasResult && t is not DrawableSpinnerHealthTick);
 
                 // tick may be null if we've hit the spin limit.
                 if (tick == null)
                 {
+                    // Find and trigger the next HP tick
+                    var hpTick = ticks.FirstOrDefault(t => !t.Result.HasResult && Time.Current >= t.StartTimeBindable.Value);
```

Comment:
what is this time check?

Reply:
Unlike regular bonus ticks, the health ticks shouldn't be triggered faster depending on the speed the player is spinning (in order for them to be evenly distributed across the post-bonus cap timeframe), therefore they are only triggered after their StartTime has elapsed.